File: LogicalThinkingPatterns/DiamondPattern/Diamond.ts
class Diamond {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = n - i - 1; j > 0; j--) {
        pattern += " ";
      }
      for (let k = 0; k <= i; k++) {
        pattern += "* ";
      }
      console.log(pattern);
    }

    for (let i = n; i > 0; i--) {
      let pattern: string = "";
      for (let j = n - i; j > 0; j--) {
        pattern += " ";
      }
      for (let k = 0; k < i; k++) {
        pattern += "* ";
      }
      console.log(pattern);
    }
  }
}

const diamond = new Diamond();
diamond.printPattern(5);

File: LogicalThinkingPatterns/DiamondPattern/Diamond.py
class Diamond:
    def printPattern(self, n):
        for i in range(0,n):
            pattern = ""
            for j in range(n - i - 1, 0, -1):
                pattern += " "
            for k in range(0, i+1):
                pattern += "* "

            print(pattern)
        
        for i in range(n,0,-1):
            pattern = ""
            for j in range(n - i, 0, -1):
                pattern += " "
            for k in range(0, i):
                pattern += "* "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

diamond = Diamond()
diamond.run()